se/se_epub_generate_toc.py

```diff
@@ -535,70 +586,71 @@ def evaluate_descendants(soup, toc_item):
 	OUTPUTS:
 	toc_item: qualified ToC item
 	"""
-	for child in soup.children:  # we expect these to be h2, h3, h4 etc
-		if isinstance(child, Tag):
-			if not toc_item.lang:
-				toc_item.lang = child.get("xml:lang") or ""
-			epub_type = child.get("epub:type") or ""
-			if not epub_type and child.name in ["h2", "h3", "h4", "h5", "h6"]:
-				# should be a label/ordinal grouping
-				child_strings = extract_strings(child)
-				if "label" in child_strings and "ordinal" in child_strings:  # quick test
-					toc_item.title_is_ordinal = True
-					# strip label
-					child_strings = regex.sub(r"<span epub:type=\"label\">(.*?)</span>", " \\1 ", child_strings)
-					# adjust roman
-					child_strings = regex.sub(r"ordinal z3998:roman", "z3998:roman", child_strings)
-					child_strings = regex.sub(r"z3998:roman ordinal", "z3998:roman", child_strings)
-					# remove ordinal
-					child_strings = regex.sub(r"<span epub:type=\"ordinal\">(.*?)</span>", " \\1 ", child_strings)
-					# remove extra spaces
-					child_strings = regex.sub(r"[ ]{2,}", " ", child_strings)
-					toc_item.title = child_strings.strip()
-				continue  # skip the following
-			if "z3998:roman" in epub_type:
-				toc_item.roman = extract_strings(child)
-				if not toc_item.title:
-					toc_item.title = f"<span epub:type=\"z3998:roman\">{toc_item.roman}</span>"
-			elif "ordinal" in epub_type:  # but not a roman numeral or a labelled item, cases caught caught above
-				if not toc_item.title:
+	children = node.xpath("./h2 | ./h3 | ./h4 | ./h5 | ./h6")
+	for child in children:  # we expect these to be h2, h3, h4 etc
+		if not toc_item.lang:
+			toc_item.lang = child.get_attr("xml:lang")
+		epub_type = child.get_attr("epub:type")
+		if not epub_type and child.tag in ["h2", "h3", "h4", "h5", "h6"]:
```

Comment:
Is the `and` here necessary? We already selected for these tags in the xpath expression